inc/server/class-form-server.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * Card server logic.
+ *
+ * @package ThemeIsle\GutenbergBlocks\Server
+ */
+
+namespace ThemeIsle\GutenbergBlocks\Server;
+
+/**
+ * Class Plugin_Card_Server
+ */
+class Form_Server {
+
+	/**
+	 * The main instance var.
+	 *
+	 * @var Form_Server
+	 */
+	public static $instance = null;
+
+	/**
+	 * Rest route namespace.
+	 *
+	 * @var Form_Server
+	 */
+	public $namespace = 'themeisle-gutenberg-blocks/';
+
+	/**
+	 * Rest route version.
+	 *
+	 * @var Form_Server
+	 */
+	public $version = 'v1';
+
+	/**
+	 * Initialize the class
+	 */
+	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register REST API route
+	 */
+	public function register_routes() {
+		$namespace = $this->namespace . $this->version;
+
+		register_rest_route(
+			$namespace,
+			'/forms',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_form_data' ),
+					'permission_callback' => function () {
+						return __return_true();
+					},
+				),
+			)
+		);
+
+		register_rest_route(
+			$namespace,
+			'/forms',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'get_form_data' ),
+					'permission_callback' => function () {
+						return __return_true();
+					},
+				),
+			)
+		);
+	}
+
+	/**
+	 * Search WordPress Plugin
+	 *
+	 * Search WordPress plugin using WordPress.org API.
+	 *
+	 * @param mixed $request Search request.
+	 *
+	 * @return mixed|\WP_REST_Response
+	 */
+	public function get_form_data( $request ) {
+
+		$return = array(
+			'success' => false,
+		);
+
+		$data = json_decode( $request->get_body(), true );
+
+		$email_subject = '[Otter] ' . ( isset( $data['emailSubject'] ) ? $data['emailSubject'] : __( 'A new submission', 'otter-blocks' ) );
+		$email_body  = $this->prepare_body( $data['data'] );
+
+		// Sent the form date to the admin site as a default behaviour.
+		$to = sanitize_email( get_site_option( 'admin_email' ) );
+		// Check if we need to send it to another user email.
+		if ( isset( $data['formOption'] ) ) {
+			$option_name = sanitize_text_field( $data['formOption'] );
+			$form_emails = get_option('themeisle_blocks_form_emails');
+
+			foreach( $form_emails as $form ) {
+				if( $form['form'] === $option_name ) {
+					$to = $form['email'];
+				}
+			}
+		}
+
+		$from    = sanitize_email( get_site_option( 'admin_email' ) );
+		$headers = array( 'Content-Type: text/html; charset=UTF-8', 'From: Admin <' . $from . '>' );
+
+		try {
+			wp_mail( $to, $email_subject, $email_body, $headers );
+			$return['success']    = true;
+		} catch ( \Exception $e ) {
+			$return['error'] = $e->getMessage();
+		} finally {
+			return rest_ensure_response( $return );
+		}
+	}
+
+	/**
+	 * Body template preparation
+	 *
+	 * @param array $data Data from the forms.
+	 *
+	 * @return string
+	 */
+	private function prepare_body( $data ) {
```

Comment:
We should include the page link in the body, a link like:

Form submission from: https://example.com/contact